iset.mm

```diff
@@ -155556,6 +155607,53 @@ WLPO is known to not be provable in IZF (and most constructive
       VLVMLNVNVRWEWFVJVOLNVPVLVS $.
   $}
 
+  ${
+    $d x y $.
+    $( Real trichotomy implies decidability of real number equality.  Or in
+       other words, analytic LPO implies analytic WLPO (see ~ trilpo and
+       ~ redcwlpo ).  Thus, this is an analytic analogue to ~ lpowlpo .
+       (Contributed by Jim Kingdon, 24-Jul-2024.) $)
+    tridceq $p |- ( A. x e. RR A. y e. RR ( x < y \/ x = y \/ y < x )
+        -> A. x e. RR A. y e. RR DECID x = y ) $=
```

Comment:
I'm wondering if in all these theorems, by using symmetry by translation, we can weaken the hypothesis/antecedent to `A. x e. RR ( x < 0 \/ x = 0 \/ 0 < x )` ?

Reply:
Yes, that would work, and the pull request has it, stated in a slightly different form, as `reap0`.

Reply:
Perfect. What I mean is that this theorem is probably better stated as
```
    tridceq $p |- ( A. x e. RR ( x < 0 \/ x = 0 \/ 0 < x )
        -> A. x e. RR A. y e. RR DECID x = y ) $=
```
since the antecedent is both simpler and a priori weaker (it only turns out to be equivalent after a bit of thought).

Reply:
Since the various choices here are equivalent (and in many cases we have proved them to be equivalent), this is for the most part a potentially hard to resolve style question.

At the risk of getting more into notation topics that I might prefer, I'll list what we have now:

Regular omniscience principles:

* `_om e. Omni` (LPO)
* `_om e. WOmni` (WLPO)
* `_om e. Markov` (MP)
* proposed for LLPO at #4058 

Analytic omniscience principles:

* `∀𝑥 ∈ ℝ ∀𝑦 ∈ ℝ (𝑥 < 𝑦 ∨ 𝑥 = 𝑦 ∨ 𝑦 < 𝑥)` (analytic LPO)
* `∀𝑥 ∈ ℝ ∀𝑦 ∈ ℝ DECID 𝑥 = 𝑦`  (analytic WLPO)
* `A. x e. RR A. y e. RR ( x =/= y -> x =//= y )` (analytic MP)
* `A. x e. RR A. y e. RR ( x <_ y \/ y <_ x )` (analytic LLPO)

Reply:
That is a nice synthetic view, which may be worth adding somewhere in iset.mm, maybe in some section header comment, with links to related definition for the regular principles, and main theorem for the analytic principles.

I think the difference between the two versions of tridceq in this thread is more than only style (two nested quantifiers is more complicated than one), but this is non-blocking, even less so since it's in a mathbox.

Reply:
> may be worth adding somewhere in iset.mm, maybe in some section header comment, with links to related definition for the regular principles, and main theorem for the analytic principles.

I've put a version of this in a branch for my next pull request.